src/utils/notification/helpers.ts

```diff
@@ -0,0 +1,225 @@
+      return preferences.videoCommentCreatedNotificationEnabled
+    case 'VideoLiked':
+      return preferences.videoLikedNotificationEnabled
+    case 'VideoDisliked':
+      return preferences.videoDislikedNotificationEnabled
+    case 'EnglishAuctionSettled':
+      return preferences.timedAuctionExpiredNotificationEnabled
+    case 'ChannelSuspended':
+      return preferences.yppChannelSuspendedNotificationEnabled
+    case 'ChannelVerified':
+      return preferences.yppChannelVerifiedNotificationEnabled
+    case 'NftPurchased':
+      return preferences.nftBoughtNotificationEnabled
+    case 'CreatorReceivesAuctionBid':
+      return preferences.bidMadeOnNftNotificationEnabled
+    case 'RoyaltyPaid':
+      return preferences.royaltyReceivedNotificationEnabled
+    case 'DirectChannelPaymentByMember':
+      return preferences.channelPaymentReceivedNotificationEnabled
+    case 'ChannelFundsWithdrawn':
+      return preferences.channelFundsWithdrawnNotificationEnabled
+    case 'ChannelCreated':
+      return preferences.channelCreatedNotificationEnabled
+    case 'CommentReply':
+      return preferences.replyToCommentNotificationEnabled
+    case 'ReactionToComment':
+      return preferences.reactionToCommentNotificationEnabled
+    case 'VideoPosted':
+      return preferences.videoPostedNotificationEnabled
+    case 'NewAuction':
+      return preferences.newNftOnAuctionNotificationEnabled
+    case 'NewNftOnSale':
+      return preferences.newNftOnSaleNotificationEnabled
+    case 'EnglishAuctionLost':
+      return preferences.auctionLostNotificationEnabled
+    case 'EnglishAuctionWon':
+      return preferences.auctionWonNotificationEnabled
+    case 'OpenAuctionLost':
+      return preferences.auctionLostNotificationEnabled
+    case 'OpenAuctionWon':
+      return preferences.auctionWonNotificationEnabled
+    default:
+      return new NotificationPreference({ inAppEnabled: false, emailEnabled: false })
+  }
+}
+
+export async function addNotification(
+  em: EntityManager,
+  account: Flat<Account> | null,
+  notificationType: NotificationType,
+  event?: Event
+) {
+  if (account) {
+    const notificationChainTag = event ? 'OnChainNotification' : 'OffChainNotification'
+    // create notification as disabled = true
+    const { inAppEnabled, emailEnabled } = preferencesForNotification(
+      account.notificationPreferences,
+      notificationType
+    )
+    // create notification (for the notification center)
+    const nextNotificationId = await getNextIdForEntity(em, notificationChainTag)
+    const notification = new Notification({
+      id: notificationChainTag + '-' + nextNotificationId.toString(),
+      accountId: account.id,
+      notificationType,
+      eventId: event?.id,
+      status: new Unread(),
+      createdAt: new Date(),
+    })
+
+    // deliver via mail if enabled
+    if (emailEnabled) {
+      await deliverNotificationViaEmail(em, account, notification)
+    }
```

Comment:
I have set a try catch block with a `console.err` in the case `sendgrid.send` is not successful
(temporary since I guess the scheduler has now be to included into the scope for notification v1)